python/ql/src/semmle/python/dataflow/SsaDefinitions.qll

```diff
@@ -79,24 +79,12 @@ abstract class PythonSsaSourceVariable extends SsaSourceVariable {
     }
 
     override predicate hasRefinementEdge(ControlFlowNode use, BasicBlock pred, BasicBlock succ) {
+        test_contains(pred.getLastNode(), use) and
```

Comment:
Compared to what the code did previously, `test_contains` contains an additional `instanceof Expr` check. It's not immediately clear to me whether this affects the behaviour. Can you weigh in on this?

Reply:
The last node in a branching block will always be an `Expr`. So this doesn't change behaviour, just makes it faster.